Content.Server/Blob/BlobCoreSystem.cs

```diff
@@ -0,0 +1,338 @@
+            return false;
+
+        blobObserverComponent.Core = blobCoreUid;
+
+        _mindSystem.TryGetMind(userId, out var mind);
+        if (mind == null)
+            return false;
+
+        _mindSystem.TransferTo(mind, observer, ghostCheckOverride: false);
+
+        _alerts.ShowAlert(observer, AlertType.BlobHealth, (short) Math.Clamp(Math.Round(core.CoreBlobTotalHealth.Float() / 10f), 0, 20));
+
+        var antagPrototype = _prototypeManager.Index<AntagPrototype>(core.AntagBlobPrototypeId);
+        var blobRole = new BlobRole(mind, antagPrototype);
+
+        _mindSystem.AddRole(mind, blobRole);
+        SendBlobBriefing(mind);
+
+        blobRule.Blobs.Add(blobRole);
+
+        if (_prototypeManager.TryIndex<ObjectivePrototype>("BlobCaptureObjective", out var objective)
+            && objective.CanBeAssigned(mind))
+        {
+            _mindSystem.TryAddObjective(blobRole.Mind, objective);
+        }
+
+        if (_mindSystem.TryGetSession(mind, out var session))
+        {
+            _audioSystem.PlayGlobal(core.GreetSoundNotification, session);
+        }
+
+        _blobObserver.UpdateUi(observer, blobObserverComponent);
+
+        return true;
+    }
+
+    private void SendBlobBriefing(Mind.Mind mind)
+    {
+        if (_mindSystem.TryGetSession(mind, out var session))
+        {
+            _chatManager.DispatchServerMessage(session, Loc.GetString("blob-role-greeting"));
+        }
+    }
+
+    private void OnDamaged(EntityUid uid, BlobCoreComponent component, DamageChangedEvent args)
+    {
+        var maxHealth = component.CoreBlobTotalHealth;
+        var currentHealth = maxHealth - args.Damageable.TotalDamage;
+
+        if (component.Observer != null)
+            _alerts.ShowAlert(component.Observer.Value, AlertType.BlobHealth, (short) Math.Clamp(Math.Round(currentHealth.Float() / 10f), 0, 20));
+    }
+
+    private void OnStartup(EntityUid uid, BlobCoreComponent component, ComponentStartup args)
+    {
+        ChangeBlobPoint(uid, 0, component);
+
+        if (TryComp<BlobTileComponent>(uid, out var blobTileComponent))
+        {
+            blobTileComponent.Core = uid;
+            blobTileComponent.Color = component.ChemСolors[component.CurrentChem];
+            Dirty(blobTileComponent);
+        }
+
+        component.BlobTiles.Add(uid);
+
+        ChangeChem(uid, component.DefaultChem, component);
+    }
+
+    public void ChangeChem(EntityUid uid, BlobChemType newChem, BlobCoreComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+
+        if (newChem == component.CurrentChem)
+            return;
+
+        var oldChem = component.CurrentChem;
+        component.CurrentChem = newChem;
+        foreach (var blobTile in component.BlobTiles)
+        {
+            if (!TryComp<BlobTileComponent>(blobTile, out var blobTileComponent))
+                continue;
+
+            blobTileComponent.Color = component.ChemСolors[newChem];
```

Comment:
Make a helper for adjusting blob color, and have **blobtilesystem** handle it's own behavior. Blob core does not need to be doing stuff for it.